api/v1alpha1/mariadb_types.go

```diff
@@ -91,6 +91,13 @@ type Storage struct {
 	// +optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced"}
 	VolumeClaimTemplate *VolumeClaimTemplate `json:"volumeClaimTemplate,omitempty"`
+	// PersistentVolumeClaimRetentionPolicy describes the lifecycle of PVCs created from volumeClaimTemplates.
+	// By default, all persistent volume claims are created as needed and retained until manually deleted.
+	// This policy allows the lifecycle to be altered, for example by deleting PVCs when their statefulset is deleted,
+	// or when their pod is scaled down.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	PVCRetentionPolicy *appsv1.StatefulSetPersistentVolumeClaimRetentionPolicy `json:"pvcRetentionPolicy,omitempty"`
```

Comment:
We don't directly embed the Kubernetes types in our CRs, we create our own here pointing to the upstream documentation:
- https://github.com/mariadb-operator/mariadb-operator/blob/main/api/v1alpha1/kubernetes_types.go

We also have an adapter here to transform between them:
- https://github.com/mariadb-operator/mariadb-operator/tree/main/pkg/kubernetes/adapter

This is to avoid hitting the [1MB hard limit of etcd when installing helm releases](https://github.com/helm/helm/issues/1413)